From f68725ad30151dab24f8b72b58b3b6d90c6d23eb Mon Sep 17 00:00:00 2001
From: ZCShou <72115@163.com>
Date: Tue, 9 Dec 2025 16:12:38 +0800
Subject: [PATCH] fix build error for phytium

---
 bsp/phytium/aarch64/rtconfig.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/bsp/phytium/aarch64/rtconfig.h b/bsp/phytium/aarch64/rtconfig.h
index 7c313e72b2..06272ba73c 100644
--- a/bsp/phytium/aarch64/rtconfig.h
+++ b/bsp/phytium/aarch64/rtconfig.h
@@ -268,6 +268,7 @@
 
 #define RT_USING_SAL
 #define SAL_INTERNET_CHECK
+#define SOCKET_TABLE_STEP_LEN 4
 
 /* Docking with protocol stacks */
 
-- 
2.34.1

